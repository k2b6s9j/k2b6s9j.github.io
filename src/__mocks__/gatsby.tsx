import * as React from "react";
const gatsby = jest.requireActual("gatsby");
import { graphql } from "graphql";

type MockedGraphQLReturn = Promise<any>;

const mockedGraphQLQuery = (query: string): MockedGraphQLReturn =>
  graphql(undefined, query);

interface StaticQueryHOCProps {
  query: MockedGraphQLReturn;
  render?: (data: object) => React.ReactElement;
  children?: (data: object) => React.ReactElement;
}

const mockedStaticQueryHOC = (
  props: StaticQueryHOCProps
): React.ReactElement => {
  if (!props.children || !props.render) {
    throw new Error("Render prop must be provided!");
  }

  return (
    (props.children && props.children({})) ||
    (props.render && props.render({})) || <div>Error!</div>
  );
};

const mockedStaticQueryHook = (data: MockedGraphQLReturn): object => ({});

module.exports = {
  ...gatsby,
  graphql: jest.fn(mockedGraphQLQuery),
  Link: jest.fn(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      to,
      ...rest
    }): React.ReactElement => <a {...rest} href={to} />
  ),
  StaticQuery: jest.fn(mockedStaticQueryHOC),
  useStaticQuery: jest.fn(mockedStaticQueryHook)
};
