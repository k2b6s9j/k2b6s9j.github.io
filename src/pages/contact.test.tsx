import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Page from "./contact";

beforeEach(() => {
  (useStaticQuery as jest.Mock).mockImplementation(() => ({
    file: {
      childImageSharp: {
        fixed: {
          base64: "",
          width: 100,
          height: 100,
          src: "",
          srcSet: ""
        }
      }
    },
    backdrop: {
      childImageSharp: {
        fluid: {
          base64: "",
          aspectRatio: 1.333,
          src: "",
          srcSet: "",
          srcWebp: "",
          srcSetWebp: "",
          sizes: ""
        }
      }
    },
    site: {
      siteMetadata: {
        title: "",
        description: "",
        siteUrl: "",
        twitterUsername: "",
        instagramUsername: "",
        linkedinUsername: "",
        githubUsername: ""
      }
    }
  }));
});

describe("Contact Page", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Page />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
