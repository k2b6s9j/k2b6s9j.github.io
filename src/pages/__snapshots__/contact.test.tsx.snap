// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page renders correctly 1`] = `
Array [
  .emotion-7 {
  height: 3rem;
  width: 100vw;
  margin-bottom: 1em;
}

.emotion-6 {
  height: 3rem;
  max-width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1rem;
  padding-right: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-1 {
  text-align: start;
  max-width: 50%;
}

.emotion-0 {
  font-weight: 600;
  color: #141414;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-5 {
  text-align: end;
  max-width: 50%;
}

.emotion-5 > :not(:last-child) {
  margin-right: 1em;
}

<head>
    Object {}
    Object {
      "lang": "en",
    }
    <meta
      content="Kepler Sticka-Jones"
      data-react-helmet={true}
      name="author"
    />
    <meta
      content=""
      data-react-helmet={true}
      name="description"
    />
    <meta
      content="#409040"
      data-react-helmet={true}
      name="theme-color"
    />
    <meta
      content="Contact | "
      data-react-helmet={true}
      property="og:title"
    />
    <meta
      content="website"
      data-react-helmet={true}
      property="og:type"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:url"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:image"
    />
    <meta
      content="image/jpeg"
      data-react-helmet={true}
      property="og:image:type"
    />
    <meta
      content="480"
      data-react-helmet={true}
      property="og:image:width"
    />
    <meta
      content="480"
      data-react-helmet={true}
      property="og:image:height"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:description"
    />
    <meta
      content="summary"
      data-react-helmet={true}
      name="twitter:card"
    />
    <meta
      content="@"
      data-react-helmet={true}
      name="twitter:site"
    />
    <meta
      content="@"
      data-react-helmet={true}
      name="twitter:creator"
    />
    <meta
      content="E5D802A305BE2DD99EB4484A4B1DD90E"
      data-react-helmet={true}
      name="msvalidate.01"
    />
    <script
      data-react-helmet={true}
      type="application/ld+json"
    >
      Object {
      "@context": "http://www.schema.org",
      "@type": "ContactPage",
      "breadcrumb": Object {
        "@type": "BreadcrumbList",
        "itemListElement": Array [
          Object {
            "@type": "ListItem",
            "item": Object {
              "@id": "https://keplersj.com/",
              "name": "Kepler Sticka-Jones",
            },
            "position": 1,
          },
          Object {
            "@type": "ListItem",
            "item": Object {
              "@id": "https://keplersj.com/contact/",
              "name": "Contact",
            },
            "position": 2,
          },
        ],
      },
      "name": "Contact | Kepler Sticka-Jones",
      "url": "/contact",
    }
    </script>
    <title
      data-react-helmet={true}
    >
      Contact | 
    </title>
  </head>

<div
    className="emotion-7 gbi-00000000000-0000000000000000000000 gatsby-image-wrapper"
    style={
      Object {
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "opacity": 0.99,
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
          .emotion-7.gbi-00000000000-0000000000000000000000:before,
          .emotion-7.gbi-00000000000-0000000000000000000000:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 0.25s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .emotion-7.gbi-00000000000-0000000000000000000000:before {
            z-index: -100;
            
            
            opacity: 0; 
          }
          .emotion-7.gbi-00000000000-0000000000000000000000:after {
            z-index: -101;
            
            
            
          }
        ",
        }
      }
    />
    <noscript>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
          .emotion-7.gbi-00000000000-0000000000000000000000:before {
            opacity: 1;
            background-image: ;
          }",
          }
        }
      />
    </noscript>
    <nav
      className="emotion-6"
    >
      <div
        className="emotion-1"
      >
        <a
          className="emotion-0"
          href="/"
        >
          Kepler Sticka-Jones
        </a>
      </div>
      <div
        className="emotion-5"
      >
        <a
          className="emotion-0"
          href="/blog"
        >
          Blog
        </a>
        <a
          className="emotion-0"
          href="/projects"
        >
          Projects
        </a>
        <a
          className="emotion-0"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </nav>
  </div>,
  .emotion-7 {
  min-height: 100vh;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 3rem 1.5rem;
}

.emotion-5 {
  text-align: center !important;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 0 auto;
  position: relative;
  width: auto;
}

@media screen and (min-width:1024px) {
  .emotion-5 {
    max-width: 960px;
  }
}

@media screen and (min-width:1216px) {
  .emotion-5 {
    max-width: 1152px;
  }
}

.emotion-0 {
  font-size: 3rem;
  font-weight: 600;
  line-height: 1.125;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1.5rem;
}

.emotion-2 {
  font-size: 2rem;
  font-weight: 600;
  line-height: 1.125;
}

.emotion-2:not(:last-child) {
  margin-bottom: 1.5rem;
}

.emotion-1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: inset 0 -1px 0 0 rgba(0,0,0,0.5);
}

.emotion-1:hover {
  box-shadow: inset 0 -2px 0 0 rgba(0,0,0,0.7);
}

@media (prefers-color-scheme:dark) {
  .emotion-1 {
    box-shadow: inset 0 -1px 0 0 rgba(255,255,255,0.5);
  }

  .emotion-1:hover {
    box-shadow: inset 0 -2px 0 0 rgba(255,255,255,0.7);
  }
}

<main>
    <section
      className="emotion-7"
    >
      <div
        className="emotion-6"
      >
        <div
          className="emotion-5"
        >
          <h1
            className="emotion-0"
          >
            Contact
          </h1>
          <h3
            className="emotion-2"
          >
            Email:
             
            <a
              className="emotion-1"
              href="mailto:kepler@stickajones.org"
            >
              kepler@stickajones.org
            </a>
          </h3>
          <h3
            className="emotion-2"
          >
            Public Key:
             
            <a
              className="emotion-1"
              href="https://keybase.io/keplersj/pgp_keys.asc?fingerprint=bd03eb5926203455cf0a838ef711c2d538d9cf6c"
            >
              BD03 EB59 2620 3455 CF0A 838E F711 C2D5 38D9 CF6C
            </a>
          </h3>
        </div>
      </div>
    </section>
  </main>,
]
`;
